src/Sil/Component/Address/Repository/CityRepositoryInterface.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Address\Repository;
+
+use InvalidArgumentException;
+use Blast\Component\Resource\Repository\ResourceRepositoryInterface;
+use Doctrine\Common\Collections\Collection;
+use Sil\Component\Address\Model\CityInterface;
+
+interface CityRepositoryInterface extends ResourceRepositoryInterface
+{
+    /**
+     * Retreive the city for specific code.
+     *
+     * @param string $code
+     *
+     * @return CityInterface
+     *
+     * @throws InvalidArgumentException
+     */
+    public function getByCode(string $code): CityInterface;
```

Comment:
findByCode ?

Reply:
Diffrence between `get` and `find` would be : find tries to find an element, so the method would return a model or a null result, a get must return a model. If not, it's an Exception (InvalidargumentException usualy)